apps/src/gamelab/GameLab.js

```diff
@@ -415,40 +427,73 @@ GameLab.prototype.onPuzzleComplete = function (submit) {
   // Stop everything on screen
   this.reset();
 
-  if (this.testResults >= this.studioApp_.TestResults.FREE_PLAY) {
-    this.studioApp_.playAudio('win');
-  } else {
-    this.studioApp_.playAudio('failure');
-  }
-
   var program;
+  var containedLevelInfo;
 
-  if (this.level.editCode) {
-    // If we want to "normalize" the JavaScript to avoid proliferation of nearly
-    // identical versions of the code on the service, we could do either of these:
+  if (this.getContainedLevelResults) {
+    var results = this.getContainedLevelResults();
+    if (results.length !== 1) {
+      throw "Exactly one contained level result is currently required.";
+    }
+    var containedResult = results[0].result;
```

Comment:
results[0] is repeated enough that maybe worth yanking into its own variable, though unfortunate that it contains a result inside itself.